src/header/internals/item.rs

```diff
@@ -51,7 +54,7 @@ impl Item {
         &raw[..]
     }
 
-    pub fn typed<H: Header + HeaderFormat>(&self) -> Option<&H> {
+    pub fn typed<H: Header + HeaderFormat + Any>(&self) -> Option<&H> {
```

Comment:
This is totally a judgement call, but do you think we should add `Any` to the `HeaderFormat` supertraits?